File: UVA/AOAPC-BAC/Ch3/Example/UVA-272_TEX Quotes.py
"""
    tags: string, input, 紫書-Ch3, CPE49
"""
flag = 0

while (True):
    try:
        line = input()
    except EOFError:
        break

    for ch in line:
        if ch == "\"":
            print("``" if flag == 0 else "''", end="")
            flag = flag ^ 1
        else:
            print(ch, end="")
    print()

File: UVA/AOAPC-BAC/Ch3/Example/UVA-272_TEX Quotes.cpp
/*
    tags: string, input, 紫書-Ch3, CPE49
*/
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    string s;
    int flag = 0;
    while (getline(cin, s)) {
        for (auto &ch: s){
            if (ch == '"') {
                cout << ((flag == 0)? "``" : "''");
                flag = 1 - flag;
            } else {
                cout << ch;
            }
        }
        cout << endl;
    }

    return 0;
}